lib/ui/recommendation-row.js

```diff
@@ -256,8 +256,10 @@ RecommendationRow.prototype = {
     const val = key.split('.').reduce(function(o, x) {
       return (typeof o == 'undefined' || o === null) ? o : o[x];
     }, this._data);
```

Comment:
Optional suggestion: how about this alternative for the first part? It handles all possible edge cases (like trying to access object properties on an array). It's slower (using try-catch sends you into interpreted vs compiled mode), but we're calling this in the midst of dealing with the DOM anyway, so speed's not a primary concern.

```js
const val = key.split('.').reduce((o,x) => {
  let result;
  try {
    result = o[x];
  } catch (ex) {}
  return result;
}, this._data);
```

In the spirit of minimizing changes just before launch, I'm fine with ignoring this suggestion, but wanted to put it out there.